packages/magic/src/index.ts

```diff
@@ -43,7 +43,8 @@ export class Magic extends Connector {
     this.actions.startActivation()
 
     await this.startListening(configuration).catch((error: Error) => {
-      this.actions.reportError(error)
+      this.actions.resetState()
+      throw error
```

Comment:
not sure if i should throw or use `onError` here?

Reply:
throwing is fine because this is in `activate` which is user-initiated